src/main/java/salesforce/ui/pages/opportunity/opportunitycontactroles/ContactRolesLightningPage.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    /**
+     * Gets composed web element.
+     *
+     * @param xpath value.
+     * @param contactName value.
+     * @param rolName value.
+     * @return web element.
+     */
+    private WebElement getWebElementRol(final String xpath, final String contactName, final String rolName) {
+        return webDriver.findElement(By.xpath(String.format(xpath, contactName, rolName)));
+    }
+
+    /**
+     * Gets contact name.
+     *
+     * @param contactName value.
+     * @return contact name.
+     */
+    private String getContactName(final String contactName) {
+        webDriverWait.until(ExpectedConditions.visibilityOf(getWebElement(CONTACT_NAME, contactName)));
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(getWebElement(CONTACT_NAME, contactName)));
+        return getWebElement(CONTACT_NAME, contactName).getText();
+    }
+
+    /**
+     * Gets contact rol name.
+     *
+     * @param contactName value.
+     * @param rolName value.
+     * @return contact name.
+     */
+    private String getContactRole(final String contactName, final String rolName) {
+        webDriverWait.until(ExpectedConditions.visibilityOf(getWebElementRol(CONTACT_ROLE, contactName, rolName)));
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(getWebElementRol(
+                CONTACT_ROLE, contactName, rolName)));
+        return getWebElementRol(CONTACT_ROLE, contactName, rolName).getText();
+    }
+
+    /**
+     * Iterates the contacts.
+     *
+     * @param contacts value.
+     * @return contact values.
+     */
+    public HashMap<String, String> iterateContacts(final HashMap<String, String> contacts) {
+        contactValues = new HashMap<>();
+        String name;
+        String rol;
+        for (String contactName : contacts.keySet()) {
+            name = getContactName(contactName);
+            rol = getContactRole(contactName, contacts.get(contactName));
+            contactValues.put(name, rol);
+        }
+        return contactValues;
+    }
+
+    @Override
+    public HashMap<String, String> verifyContactRoles(final HashMap<String, String> contactRolList) {
+        HashMap<String, String> contactValues;
+        try {
+            webDriverWait.until(ExpectedConditions.visibilityOf(contactRolesTitle));
+            webDriverWait.until(ExpectedConditions.elementToBeClickable(contactRolesTitle));
+            contactValues = iterateContacts(contactRolList);
+        } catch (StaleElementReferenceException elementHasDisappeared) {
+            webDriverWait.until(ExpectedConditions.visibilityOf(contactRolesTitle));
+            webDriverWait.until(ExpectedConditions.elementToBeClickable(contactRolesTitle));
+            contactValues = this.iterateContacts(contactRolList);
+        }
+        return contactValues;
+    }
+
+    @Override
+    public AbstractOpportunityPage setContacts(final HashMap<String, String> contactsList) {
+        return null;
```

Comment:
It is correct?

Reply:
yes, it is returning null because the structure was no adecuated for both user experinces